src/main/java/Duke.java

```diff
@@ -1,10 +1,271 @@
+                    }
+                    printFileContents(textFilePath);
+                } catch (EmptyListException e) {
+                    System.out.println(e.getExceptionMessage());
+                } catch (FileNotFoundException f) {
+                    System.out.println("File not found.");
+                }
+                break;
+            case COMMAND_MARK:
+                int itemNumber = Integer.parseInt(line.split(" ")[1]) - 1;
+                try {
+                    if (itemNumber < 0 || itemNumber > tasksList.size() - 1) {
+                        throw new IllegalTaskNumber("this task does not exist");
+                    } else {
+                        Task toBeChanged = tasksList.get(itemNumber);
+                        if (toBeChanged.isDone) {
+                            throw new IllegalMarkingException("already marked");
+                        } else {
+                            toBeChanged.markAsDone(toBeChanged);
+                            String textToAdd = toBeChanged.toFileString();
+                            taskOverrideList.set(itemNumber, textToAdd);
+                            writeToFile(textFilePath, taskOverrideList.get(0));
+                            for (int i = 1; i < taskOverrideList.size(); i++) {
+                                appendToFile(textFilePath, taskOverrideList.get(i));
+                            }
+                        }
+                    }
+                } catch (IllegalMarkingException e) {
+                    System.out.println(e.getExceptionMessage());
+                } catch (IllegalTaskNumber f) {
+                    System.out.println(f.getExceptionMessage());
+                } catch (IOException g) {
+                    System.out.println("Something went wrong. I am unable to mark this task.");
+                }
+                break;
+            case COMMAND_UNMARK:
+                itemNumber = Integer.parseInt(line.split(" ")[1]) - 1;
+                try {
+                    if (itemNumber < 0 || itemNumber > tasksList.size() - 1) {
+                        throw new IllegalTaskNumber("this task does not exist");
+                    } else {
+                        Task toBeChanged = tasksList.get(itemNumber);
+                        if (toBeChanged.isDone) {
+                            toBeChanged.markAsUndone(toBeChanged);
+                            String textToAdd = toBeChanged.toFileString();
+                            taskOverrideList.set(itemNumber, textToAdd);
+                            writeToFile(textFilePath, taskOverrideList.get(0));
+                            for (int i = 1; i < taskOverrideList.size(); i++) {
+                                appendToFile(textFilePath, taskOverrideList.get(i));
+                            }
+                        } else {
+                            throw new IllegalUnmarkingException("still unmarked");
+                        }
+                    }
```

Comment:
Similarly, perhaps consider abstracting out this portion of code to reduce deeply nested code. Additionally, if you have repeated logic, it could also help in simplifying this portion of the code base.